awx/api/views/__init__.py

```diff
@@ -1445,6 +1445,37 @@ class CredentialInputSourceSubList(SubListCreateAttachDetachAPIView):
     parent_key = 'target_credential'
 
 
+class CredentialTypeExternalTest(SubDetailAPIView):
+    view_name = _('External Credential Type Test')
+    model = models.CredentialType
+    serializer_class = serializers.EmptySerializer
+
+    def post(self, request, *args, **kwargs):
+        obj = self.get_object()
+        credential_id = request.data.get('credential', None)
+
+        test_inputs = {}
+        if credential_id is not None:
+            cred = models.Credential.accessible_objects(request.user, 'use_role').filter(id=credential_id).first()
+            if cred is None or cred.credential_type.kind != 'external':
+                return Response({}, status=status.HTTP_400_BAD_REQUEST)
+
+            for field_name in cred.inputs:
+                test_inputs[field_name] = cred.get_input(field_name)
+
+        for key, value in request.data.get('inputs', {}).items():
+            if value != '$encrypted$':
+                test_inputs[key] = value
+
+        try:
+            obj.plugin.backend(**test_inputs)
+            return Response({}, status=status.HTTP_202_ACCEPTED)
+        except RuntimeError as exc:
+            return Response({'inputs': str(exc)}, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
If we hit a `RuntimeError` when invoking the plugin, include its message in the response data.

The UI [will display the message](https://github.com/jakemcdermott/awx/blob/credential_plugins/awx/ui/client/features/credentials/add-credentials.controller.js#L105) in the toast notification if it finds it: